libbeat/processors/add_docker_metadata/add_docker_metadata.go

```
@@ -0,0 +1,93 @@
+package add_docker_metadata
+
+import (
+	"fmt"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/libbeat/processors"
+)
+
+func init() {
+	processors.RegisterPlugin("add_docker_metadata", newDockerMetadataProcessor)
+}
+
+type addDockerMetadata struct {
+	watcher Watcher
+	fields  []string
+}
+
+func newDockerMetadataProcessor(cfg common.Config) (processors.Processor, error) {
+	return buildDockerMetadataProcessor(cfg, NewWatcher)
+}
+
+func buildDockerMetadataProcessor(cfg common.Config, watcherConstructor WatcherConstructor) (processors.Processor, error) {
+	logp.Beta("The add_docker_metadata processor is beta")
+
+	config := defaultConfig()
+
+	err := cfg.Unpack(&config)
+	if err != nil {
+		return nil, fmt.Errorf("fail to unpack the add_docker_metadata configuration: %s", err)
+	}
+
+	watcher, err := watcherConstructor(config.Host, config.TLS)
+	if err != nil {
+		panic(err)
```

Why not return an error?